[MTM-58448] Extract the Reports view into a separate plugin announcement (#1995)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/ui-c8y-1020-0-0-reports-module-as-separate-plugin.md

---
date: 2024-04-26

> title: Extract the Reports view into a separate plugin

Reviewer: ```suggestion
title: Reports page extracted into a separate plugin
```

product_area: Application enablement & solutions
change_type:
  - value: change-inv-3bw8e
    label: Announcement
component:
  - value: component-YbYJ3gLU_
    label: Web SDK
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-58448
version: 1020.0.0
---

> In the upcoming version we will extract the Reports module from the Cockpit application and add it as a separate plugin.

Reviewer: ```suggestion
In the upcoming version the report module is extracted from the Cockpit application and added as a separate plugin.
```

> In the upcoming version, we will extract Reports module from Cockpit application and we will add is as a separate plugin.

Reviewer 2: Besides one spelling, maybe:
```suggestion
In the upcoming version we will extract the Reports module from the Cockpit application and add it as a separate plugin.
```
?

> This is just first step towards removing Reports completely and replace is with Dashboard manager.

Reviewer 2: ```suggestion
This is just a first step towards removing Reports completely and replacing it with Dashboard Manager.
```
?

> This is just a first step towards removing Reports completely and replacing it with Dashboard Manager.

Reviewer: ```suggestion
This is a first step towards removing the reports module entirely and replacing it with the dashboard manager.
```

> It might be breaking change as Reports module won't be part of `@c8y/ngx-components/context-dashboard` anymore, but will be part of `@c8y/ngx-components/report-dashboard`.

Reviewer: ```suggestion
This change might be a breaking change as the reports module will no longer be part of `@c8y/ngx-components/context-dashboard`, but will be part of `@c8y/ngx-components/report-dashboard`.
```

> This change won't affect visual and UX of Cockpit application nor add any new functionalities.

Reviewer: ```suggestion
This change does not affect the layout and UX of the Cockpit application nor does it add any new functionalities.
```

> As side effect, performance improvement might be noticed as Reports from now on will be loaded lazily (only when user navigates to Reports).

Reviewer: ```suggestion
As a side effect, a performance improvement might be noticed as reports from now on will be loaded lazily (that is, only when users navigate to the **Reports** page).
```
